kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -358,7 +358,12 @@ private static Optional<Metadata> assignColumnIdAndPhysicalName(
     // If this becomes hotspot, we can consider updating the methods to pass around AtomicBoolean
     // to track if the schema has changed. It is a bit convoluted to pass around and update the
     // AtomicBoolean in the recursive and multiple methods.
-    if (oldSchema.equals(newSchema)) {
+    boolean schemasEqual = oldSchema.equals(newSchema);
+    // If IDs are prepopulated on a schema then this value might not have been put in metadata yet
+    // so ensure we write the values in this case.
+    boolean maxIdPresent =
```

Comment:
If a new column is added (with all cm info present in the new column) where do we update the max column id? Maybe @amogh-jahagirdar do you know?

Reply:
Because then schemaEquals=true here, but the CM ID in the metadata != `maxColumnId` I think?

Reply:
https://github.com/delta-io/delta/pull/4364/files/cc7dbb484d4ba50f4fe5b9a350bd67066aa8aa1c#diff-38491854ca9c7c7860acd852eef7e8477196d0c9690ea55ed1691ada0c8bf631R345 will increment, I'm would need to double check to see if we account for a field added with a value already present.

Reply:
It doesn't look like we do this.  Without these values we see it updated in ` assignColumnIdAndPhysicalNameToField` and there should probably be a corresponding check.  And as in your other PR we should probably change the check to ensure maxId is written if it is greater thenwhat is in metadata.